cli/azd/cmd/env.go

```diff
@@ -150,6 +168,309 @@ func (e *envSetAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+
+	if !willCreateNewSecret {
+		// reassign messages for selecting existing secret
+		pickSubscription = "Select the subscription where the Key Vault secret is"
+		pickKvAccount = "Select the Key Vault where the Key Vault secret is"
+	}
+
+	subId, err := e.prompter.PromptSubscription(ctx, pickSubscription)
+	if err != nil {
+		return nil, fmt.Errorf("prompting for subscription: %w", err)
+	}
+	tenantId, err := e.subResolver.LookupTenant(ctx, subId)
+	if err != nil {
+		return nil, fmt.Errorf("looking up tenant for subscription: %w", err)
+	}
+
+	e.console.ShowSpinner(ctx, "Finding Key Vaults from the selected subscription", input.Step)
+	vaultsList, err := e.kvService.ListSubscriptionVaults(ctx, subId)
+	if err != nil {
+		return nil, fmt.Errorf("getting the list of Key Vaults: %w", err)
+	}
+	// prompt for vault selection
+	e.console.StopSpinner(ctx, "", input.Step)
+
+	atLeastOneKvAccountExists := len(vaultsList) > 0
+	if !atLeastOneKvAccountExists && !willCreateNewSecret {
+		e.console.MessageUxItem(ctx, &ux.WarningMessage{
+			Description: "No Azure Key Vaults were found in the selected subscription",
+		})
+		// update the flow to offer creating a new Key Vault
+		willCreateNewSecret = true
+	}
+
+	createNewKvAccountOption := "Create a new Key Vault"
+	selectKvAccountOptions := []string{}
+	// indexOffset makes the ids to start from 1 instead of 0 when displaying the options
+	indexOffset := 1
+	if willCreateNewSecret {
+		selectKvAccountOptions = append(selectKvAccountOptions, createNewKvAccountOption)
+	}
+
+	for index, vault := range vaultsList {
+		if listWithoutNumbers {
+			selectKvAccountOptions = append(selectKvAccountOptions, vault.Name)
+		} else {
+			selectKvAccountOptions = append(selectKvAccountOptions, fmt.Sprintf("%2d. %s", index+indexOffset, vault.Name))
+		}
+	}
+
+	kvAccountSelectionIndex, err := e.console.Select(ctx, input.ConsoleOptions{
+		Message:      pickKvAccount,
+		Options:      selectKvAccountOptions,
+		DefaultValue: selectKvAccountOptions[0],
+	})
+	if err != nil {
+		return nil, fmt.Errorf("selecting Key Vault: %w", err)
+	}
+
+	willCreateNewKvAccount := selectKvAccountOptions[kvAccountSelectionIndex] == createNewKvAccountOption
+	if willCreateNewSecret && !willCreateNewKvAccount {
+		// when willCreateNewSecret is true, we added a new option at the beginning of the list
+		// to recover the original kv account name
+		kvAccountSelectionIndex--
+	}
+
+	var kvAccount keyvault.Vault
+	if atLeastOneKvAccountExists {
+		kvAccount = vaultsList[kvAccountSelectionIndex]
+	}
+
+	if willCreateNewKvAccount {
+		location, err := e.prompter.PromptLocation(ctx, subId, "Select the location to create the Key Vault", nil)
+		if err != nil {
+			return nil, fmt.Errorf("prompting for Key Vault location: %w", err)
+		}
+		rg, err := e.prompter.PromptResourceGroupFrom(ctx, subId, location, prompt.PromptResourceGroupFromOptions{
+			DefaultName: "rg-for-my-kv-account",
```

Comment:
This reads closer to a placeholder value. I'm wondering if we could either suggest `rg-<project name>-kv`, or just avoid defaulting altogether.

Reply:
I didn't want to use the project name because it will be common to use one Key Vault for multiple projects.  This is a personal Key Vault that becomes handy cross-projects, if customer wants to.

We did remove the concept of a key vault account, so I do need to remove that part from here.